src/mewbot/loader.py

```diff
@@ -72,35 +78,30 @@ def load_behaviour(config: BehaviourConfigBlock) -> BehaviourInterface:
 
 
 def load_component(config: ConfigBlock) -> Component:
+    """Creates a component based on a configuration block"""
+
     # Ensure that the object we have been passed contains all required fields.
     if not _REQUIRED_KEYS.issubset(config.keys()):
         raise ValueError(
             f"Config missing some keys: {_REQUIRED_KEYS.difference(config.keys())}"
         )
 
-    # Load the module the component is expected to be in.
-    # This happens first as it may cause API versions to be registered for the first time.
-    if config["module"] not in sys.modules:
-        importlib.import_module(config["module"])
-
-    if not ComponentKind.has_value(config["kind"]):
-        raise ValueError(f"Invalid component kind {config['kind']}")
-
-    kind = ComponentKind(config["kind"])
-    interface = ComponentKind.interface(kind)
-
-    module = sys.modules[config["module"]]
+    # Identify the kind of component we should be loading, and the interface that implies.
```

Comment:
Indeed - absent very, very weird edge cases that we can probably safely ignore, this inference should always be safe.